src/ert/analysis/_es_update.py

```diff
@@ -175,11 +175,17 @@ def _load_observations_and_responses(
         # Exit early if some observations are pointing to non-existing responses
         raise ErtAnalysisError("No active observations for update step") from e
 
-    responses = observations_and_responses.responses()
-    observations = observations_and_responses.observations()
-    errors = observations_and_responses.errors()
-    observation_keys = observations_and_responses.observation_keys()
-    indexes = observations_and_responses.index()
+    responses = observations_and_responses[
+        [
+            i
+            for i in range(ensemble.ensemble_size + 1)
+            if i in observations_and_responses
+        ]
+    ].to_numpy()
+    observations = observations_and_responses["observations"].to_numpy()
+    errors = observations_and_responses["errors"].to_numpy()
+    observation_keys = observations_and_responses["obs_name"].to_numpy()
```

Comment:
Should we use observation `keys` or `names`?

Reply:
Unfortunately now we are currently using both, but we should def switch over to using only one. `*_name` convention is currently used in datasets, but we could use `*_key` (or `*_name`) uniformly for variable names

Reply:
Made an issue on it: https://github.com/equinor/ert/issues/8115